catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreAutoPublisher.java

```diff
@@ -0,0 +1,146 @@
+
+            SourceMonitor storeRegisteredSourceMonitor = new SourceMonitor() {
+                @Override
+                public void setAvailable() {
+                    if (registryStore.isAutoPush()) {
+                        registryPublish(registryStore);
+                    }
+                }
+
+                @Override
+                public void setUnavailable() {
+                }
+            };
+
+            if (registryStore.isAvailable(storeRegisteredSourceMonitor) && registryStore.isAutoPush()) {
+                if(previousState == null || !previousState) {
+                    registryPublish(registryStore);
+                }
+            }
+        }
+    }
+
+    void registryPublish(RegistryStore registryStore) {
+
+        if (!registryStore.getRegistryId()
+                .isEmpty()) {
+            try {
+
+                Optional<Metacard> registryIdentityMetacardOpt =
+                        Security.runAsAdminWithException(() -> federationAdminService.getLocalRegistryIdentityMetacard());
+
+                if (registryIdentityMetacardOpt.isPresent()) {
+                    Metacard registryIdentityMetacard = registryIdentityMetacardOpt.get();
+                    String localRegistryId = registryIdentityMetacard.getAttribute(
+                            RegistryObjectMetacardType.REGISTRY_ID)
+                            .getValue()
+                            .toString();
+                    Security.runAsAdminWithException(() -> {
+                        registryPublicationService.publish(localRegistryId,
+                                registryStore.getRegistryId());
+                        return null;
+                    });
+                }
+            } catch (Exception e) {
+                LOGGER.error("Failed to auto-push registry configuration from {}, {}",
+                        registryStore.getRegistryId(),
+                        e);
+            }
+        }
+    }
+
+    BundleContext getBundleContext() {
+        Bundle bundle = FrameworkUtil.getBundle(this.getClass());
+        if (bundle != null) {
+            return bundle.getBundleContext();
+        }
+        return null;
+    }
+
+    @Override
+    public void handleEvent(Event event) {
+        if (((ServiceEvent) event.getProperty(EventConstants.EVENT)).getType()
+                != ServiceEvent.MODIFIED) {
+            return;
+        }
+        String pid = event.getProperty(Constants.SERVICE_PID)
+                .toString();
+
+        BundleContext bundleContext = getBundleContext();
+
+        RegistryStore registryStore =
+                (RegistryStore) bundleContext.getService(((ServiceEvent) event.getProperty(
+                        EventConstants.EVENT)).getServiceReference());
+
+        Boolean isAutoPush = registryStoreMap.get(pid);
+
+        if (isAutoPush != null && !isAutoPush && registryStore.isAutoPush()) {
+            registryPublish(registryStore);
```

Comment:
The autoPush action is kind of strange because once this flag is set, there's no going back (once it's set, turning it off doesn't unpublish). Maybe we should consider changing up the wording here and adding a condition here that unpublishes the site after the flag is turned off.